File: PetShop.java
import java.util.ArrayList;
import java.util.Scanner;

class PetShop {
    private static class Product {
        private int id;
        private String name;
        private String category;
        private double price;

        public Product(int id, String name, String category, double price) {
            this.id = id;
            this.name = name;
            this.category = category;
            this.price = price;
        }

        public int getId() {
            return id;
        }

        public String getName() {
            return name;
        }

        public String getCategory() {
            return category;
        }

        public double getPrice() {
            return price;
        }

        public void setName(String name) {
            this.name = name;
        }

        public void setCategory(String category) {
            this.category = category;
        }

        public void setPrice(double price) {
            this.price = price;
        }

        @Override
        public String toString() {
            return "ID: " + id + ", Nama: " + name + ", Kategori: " + category + ", Harga: " + price;
        }
    }

    private ArrayList<Product> products = new ArrayList<>();
    private int nomor = 1;

    public void addProduct(String name, String category, double price) {
        products.add(new Product(nomor++, name, category, price));
    }

    public void displayProducts() {
        if (products.isEmpty()) {
            System.out.println("Tidak ada produk dalam daftar.");
        } else {
            for (Product product : products) {
                System.out.println(product);
            }
        }
    }

    public void updateProduct(int id, String newName, String newCategory, double newPrice) {
        for (Product product : products) {
            if (product.getId() == id) {
                product.setName(newName);
                product.setCategory(newCategory);
                product.setPrice(newPrice);
                System.out.println("Produk berhasil diperbarui.");
                return;
            }
        }
    }

    public void deleteProduct(int id) {
        boolean removed = products.removeIf(product -> product.getId() == id);
        if (removed) {
            System.out.println("Produk berhasil dihapus.");
        } else {
            System.out.println("Produk dengan ID " + id + " tidak ditemukan.");
        }
    }

    public void searchProduct(String name) {
        boolean found = false;
        for (Product product : products) {
            if (product.getName().equalsIgnoreCase(name)) {
                System.out.println(product);
                found = true;
            }
        }
        if (!found) {
            System.out.println("Produk tidak ditemukan.");
        }
    }

    public static void main(String[] args) {
        PetShop shop = new PetShop();
        Scanner scanner = new Scanner(System.in);
        
        while (true) {
            System.out.println("\nMenu:");
            System.out.println("1. Tampilkan Produk");
            System.out.println("2. Tambah Produk");
            System.out.println("3. Ubah Produk");
            System.out.println("4. Hapus Produk");
            System.out.println("5. Cari Produk");
            System.out.println("6. Keluar");
            System.out.print("Pilih menu: ");
            int choice = scanner.nextInt();
            scanner.nextLine(); 

            switch (choice) {
                case 1:
                    shop.displayProducts();
                    break;
                case 2:
                    System.out.print("Nama Produk: ");
                    String name = scanner.nextLine();
                    System.out.print("Kategori Produk: ");
                    String category = scanner.nextLine();
                    System.out.print("Harga Produk: ");
                    double price = scanner.nextDouble();
                    shop.addProduct(name, category, price);
                    break;
                case 3:
                    System.out.print("Masukkan ID Produk yang akan diubah: ");
                    int updateId = scanner.nextInt();
                    scanner.nextLine();

                    boolean exists = false;
                    for (Product product : shop.products) {
                        if (product.getId() == updateId) {
                            exists = true;
                            break;
                        }
                    }

                    if (!exists) {
                        System.out.println("Produk dengan ID " + updateId + " tidak ditemukan.");
                        break;
                    }

                    System.out.print("Nama Baru: ");
                    String newName = scanner.nextLine();
                    System.out.print("Kategori Baru: ");
                    String newCategory = scanner.nextLine();
                    System.out.print("Harga Baru: ");
                    double newPrice = scanner.nextDouble();
                    shop.updateProduct(updateId, newName, newCategory, newPrice);
                    break;
                case 4:
                    System.out.print("Masukkan ID Produk yang akan dihapus: ");
                    int deleteId = scanner.nextInt();
                    shop.deleteProduct(deleteId);
                    break;
                case 5:
                    System.out.print("Masukkan Nama Produk yang dicari: ");
                    String searchName = scanner.nextLine();
                    shop.searchProduct(searchName);
                    break;
                case 6:
                    System.out.println("Keluar dari program.");
                    scanner.close();
                    return;
                default:
                    System.out.println("Pilihan tidak valid. Coba lagi.");
            }
        }
    }
}

File: PetShop.py
class Product:
    def __init__(self, product_id, name, category, price):
        self.product_id = product_id
        self.name = name
        self.category = category
        self.price = price

    def __str__(self):
        return f"ID: {self.product_id}, Nama: {self.name}, Kategori: {self.category}, Harga: {self.price}"

class PetShop:
    def __init__(self):
        self.products = []
        self.nomor = 1

    def add_product(self, name, category, price):
        self.products.append(Product(self.nomor, name, category, price))
        self.nomor += 1

    def display_products(self):
        if not self.products:
            print("Tidak ada produk dalam daftar.")
        else:
            for product in self.products:
                print(product)

    def update_product(self, product_id, new_name, new_category, new_price):
        for product in self.products:
            if product.product_id == product_id:
                product.name = new_name
                product.category = new_category
                product.price = new_price
                print("Produk berhasil diperbarui.")
                return
        print(f"Produk dengan ID {product_id} tidak ditemukan.")

    def delete_product(self, product_id):
        for product in self.products:
            if product.product_id == product_id:
                self.products.remove(product)
                print("Produk berhasil dihapus.")
                return

    def search_product(self, name):
        found = False
        for product in self.products:
            if product.name.lower() == name.lower():
                print(product)
                found = True
        if not found:
            print("Produk tidak ditemukan.")

def main():
    shop = PetShop()

    while True:
        print("\nMenu:")
        print("1. Tampilkan Produk")
        print("2. Tambah Produk")
        print("3. Ubah Produk")
        print("4. Hapus Produk")
        print("5. Cari Produk")
        print("6. Keluar")
        choice = input("Pilih menu: ")

        if choice == "1":
            shop.display_products()

        elif choice == "2":
            name = input("Nama Produk: ")
            category = input("Kategori Produk: ")
            price = float(input("Harga Produk: "))
            shop.add_product(name, category, price)

        elif choice == "3":
            product_id = int(input("Masukkan ID Produk yang akan diubah: "))

            exists = any(product.product_id == product_id for product in shop.products)

            if not exists:
                print(f"Produk dengan ID {product_id} tidak ditemukan.")

            else:
                new_name = input("Nama Baru: ")
                new_category = input("Kategori Baru: ")
                new_price = float(input("Harga Baru: "))
                shop.update_product(product_id, new_name, new_category, new_price)


        elif choice == "4":
            product_id = int(input("Masukkan ID Produk yang akan dihapus: "))
            shop.delete_product(product_id)

        elif choice == "5":
            search_name = input("Masukkan Nama Produk yang dicari: ")
            shop.search_product(search_name)

        elif choice == "6":
            print("Keluar dari program.")
            break
        else:
            print("Pilihan tidak valid. Coba lagi.")

if __name__ == "__main__":
    main()
